.circleci/config.yml

```diff
@@ -1,23 +1,458 @@
+      NOTIFICATION_SERVICE_IDENTIFIER: 'com.mattermost.rn.NotificationService'
+      IOS_APP_GROUP: 'group.com.mattermost'
+      IOS_ICLOUD_CONTAINER: 'iCloud.com.mattermost.rn'
+      IOS_BUILD_EXPORT_METHOD: 'app-store'
+      AWS_FOLDER_NAME: 'mattermost-mobile'
+      MATTERMOST_WEBHOOK_URL: ${MATTERMOST_WEBHOOK_BUILD}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost.ipa"
+
+  build-ios-pr:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BUILD_PR: true
+      BRANCH_TO_BUILD: ${CIRCLE_BRANCH}
+      REPLACE_ASSETS: false
+      MAIN_APP_IDENTIFIER: "com.mattermost.rnbeta"
+      SENTRY_ENABLED: false
+      MATCH_TYPE: adhoc
+      MATCH_APP_IDENTIFIER: 'com.mattermost.rnbeta.NotificationService,com.mattermost.rnbeta.MattermostShare,com.mattermost.rnbeta'
+      EXTENSION_APP_IDENTIFIER: 'com.mattermost.rnbeta.MattermostShare'
+      NOTIFICATION_SERVICE_IDENTIFIER: 'com.mattermost.rnbeta.NotificationService'
+      IOS_APP_GROUP: 'group.com.mattermost.rnbeta'
+      IOS_ICLOUD_CONTAINER: 'iCloud.com.mattermost.rnbeta'
+      IOS_BUILD_EXPORT_METHOD: 'ad-hoc'
+      AWS_FOLDER_NAME: 'mattermost-mobile-pr'
+      MATTERMOST_WEBHOOK_URL: ${MATTERMOST_WEBHOOK_PR}
+    steps:
+      - build-ios
+      - persist
+      - save:
+          filename: "Mattermost_Beta.ipa"
+
+  deploy-android-release:
+    executor:
+      name: android
+      resource_class: medium
+    environment:
+      APP_NAME: "Mattermost"
+      BETA_BUILD: false
+    steps:
+      - deploy-android:
+          apk_path: Mattermost.apk
+
+  deploy-android-beta:
+    executor:
+      name: android
+      resource_class: medium
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BETA_BUILD: true
+    steps:
+      - deploy-android:
+          apk_path: Mattermost_Beta.apk
+
+  deploy-ios-release:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost"
+      BETA_BUILD: false
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost.ipa
+
+  deploy-ios-beta:
+    executor: ios
+    environment:
+      APP_NAME: "Mattermost Beta"
+      BETA_BUILD: true
+    steps:
+      - deploy-ios:
+          ipa_path: Mattermost_Beta.ipa
 
 workflows:
   version: 2
-  pr-test:
+  build:
     jobs:
       - test
+
+      - build-android-release:
+          context: mattermost-mobile
+          requires:
+            - test
+          filters:
+            branches:
+              only: /^build-(?!ios|pr|number).*/
+      - deploy-android-release:
+          context: mattermost-mobile
+          requires:
+            - build-android-release
+          filters:
+            branches:
+              only: /^build-(?!ios|pr|number).*/
+
+      - build-android-beta:
+          context: mattermost-mobile
+          requires:
+            - test
+          filters:
+            branches:
+              only: /^build-(?!ios|pr|number).*/
```

Comment:
are we building both release and beta at the same time?

shouldn't they be different?

Reply:
Yes at the same time, and I would rather build both at the same time to gain time.. there is no point to build them separately

Reply:
just to make sure I'm not missing something, wouldn't beta and release have a different timeframe? or are we saving time so it's already there and we don't have to build again later?

Reply:
We run both builds at the same time.

Today we just need to trigger one wait and then trigger the other one as we have only one box, by using circleCI we can trigger them at the same time